pkg/util/testing/core.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+func IndexedJob(name, ns string) batchv1.JobTemplateSpec {
+	indexedCompletionMode := batchv1.IndexedCompletion
+	return batchv1.JobTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+		},
+		Spec: batchv1.JobSpec{
+			CompletionMode: &indexedCompletionMode,
```

Comment:
use the "k8s.io/utils/pointer" pkg instead

Reply:
I wasn't able to find an applicable function in the pointer package, but looking at examples it seems usually either the current method I have is used, or a `completionModePtr` helper function is implemented to make it a bit clean, so I've updated it to use this approach. Let me know if you have any thoughts.

Reply:
sg